### PR TITLE
CLN: Move _convert_to_list_like to common

### PR DESCRIPTION
**Pull request by dsaxton**

### CODE REVIEW

File: pandas/core/common.py [hunk 1]
```diff
@@ -473,3 +475,14 @@ def f(x):
         f = mapper
 
     return f
+
+
+def convert_to_list_like(values):
+    if hasattr(values, "dtype"):
```
Comment (jreback) [inline pandas/core/common.py:481]: can u type and add a doc string
make these elif 

see where else we can u this

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/common.py [hunk 2]
```diff
@@ -473,3 +477,16 @@ def f(x):
         f = mapper
 
     return f
+
+
+def convert_to_list_like(
+    values: Union[Scalar, Iterable, AnyArrayLike]
+) -> Union[List, AnyArrayLike]:
```
Comment (dsaxton) [inline pandas/core/common.py:484]: Had to rewrite this a bit to get it to type check

File: pandas/core/common.py [hunk 3]
```diff
@@ -4,17 +4,16 @@
 Note: pandas.core.common is *not* part of the public API.
 """
 
-import collections
-from collections import abc
+from collections import abc, defaultdict
```
Comment (dsaxton) [inline pandas/core/common.py:7]: Slightly unrelated, but this seems better than mixing the two import styles

Comment (jbrockmendel) [reply pandas/core/common.py:7]: agreed

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
hmm azure doesnt' seem to be running. can you merge master and ping on green.

**Comment by jreback** [comment]
thanks

File: pandas/core/common.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/core/common.py:484]: @dsaxton i noticed you used ABCIndex instead of ABCIndexClass, was that intentional?

Comment (dsaxton) [reply pandas/core/common.py:484]: I don't think so, definitely change if you feel it's not correct

**Pull request merged**